extensions/wikia/TemplateClassification/services/UserTemplateClassificationService.class.php

```diff
@@ -43,4 +50,23 @@ public function getType( $wikiId, $pageId ) {
 
 		return $templateType;
 	}
+
+	/**
+	 * Verify template type before user classification
+	 *
+	 * @param int $wikiId
+	 * @param int $pageId
+	 * @param string $templateType
+	 * @param string $origin
+	 * @throws BadRequestApiException
+	 */
+	public function classifyTemplate( $wikiId, $pageId, $templateType, $origin ) {
+		if ( !in_array( $templateType, self::$templateTypes ) ) {
+			throw new ApiException( self::CLASSIFY_TEMPLATE_EXCEPTION_MESSAGE, self::CLASSIFY_TEMPLATE_EXCEPTION_CODE );
+		}
+
+		parent::classifyTemplate( $wikiId, $pageId, $templateType, self::USER_PROVIDER, $origin );
+
+		wfRunHooks( 'UserTemplateClassification::TemplateClassified', [ $wikiId, $pageId, $templateType ] );
```

Comment:
My assumption here is if we don't need Title instance in this method we shouldn't create it here. By passing `$pageId` you are able to create Title wherever you want. Also after removing methods from this hook we will not have leftovers like creating unnecessary Title instance on every classification.